dipy/denoise/enhancement_kernel.pyx

```diff
@@ -0,0 +1,420 @@
+                              DWI-MRI. J Math Imaging Vis, 46(3):326-368.
+        [Portegies2015] J. Portegies, G. Sanguinetti, S. Meesters, and R. Duits.
+                        (2015) New Approximation of a Scale Space Kernel on SE(3) 
+                        and Applications in Neuroimaging. Fifth International
+                        Conference on Scale Space and Variational Methods in
+                        Computer Vision
+        [Portegies2015b] J. Portegies, R. Fick, G. Sanguinetti, S. Meesters, 
+                         G. Girard, and R. Duits. (2015) Improving Fiber 
+                         Alignment in HARDI by Combining Contextual PDE flow with 
+                         Constrained Spherical Deconvolution. PLoS One.
+        """
+
+        # save parameters as class members
+        self.D33 = D33
+        self.D44 = D44
+        self.t = t
+
+        # define a sphere
+        if type(orientations) is Sphere:
+            # use the sphere defined by the user
+            sphere = orientations
+        elif type(orientations) is int:
+            # electrostatic repulsion based on number of orientations
+            n_pts = orientations
+            theta = np.pi * np.random.rand(n_pts)
+            phi = 2 * np.pi * np.random.rand(n_pts)
+            hsph_initial = HemiSphere(theta=theta, phi=phi)
+            sphere, potential = disperse_charges(hsph_initial , 5000)
+        else:
+            # use default
+            sphere = get_sphere('repulsion100')
+        self.orientations = sphere.vertices
+        self.sphere = sphere
+
+        # save openmp settings
+        self.num_threads = num_threads
+        
+        # file location of the lut table for saving/loading
+        kernellutpath = "%s/kernel_d33@%4.2f_d44@%4.2f_t@%4.2f_numverts%d.npy" \
+                        % (gettempdir(), D33, D44, t, len(self.orientations))
+
+        # create a lookup table in testing mode
+        if test_mode:
+            self.create_lookup_table(True)
+            return
+
+        # if LUT exists, load
+        if not force_recompute and os.path.isfile(kernellutpath):
+            print "The kernel already exists. Loading from " + kernellutpath
+            self.lookuptable = np.load(kernellutpath)
+
+        # else, create
+        else:
+            print "The kernel doesn't exist yet. Computing..."
+            self.create_lookup_table()
+            np.save(kernellutpath, self.lookuptable)
+            
+    def get_lookup_table(self):
+        """ Return the computed look-up table.
+        """
+        return self.lookuptable
+
+    def get_orientations(self):
+        """ Return the orientations.
+        """
+        return self.orientations
+        
+    def get_sphere(self):
+        """ Get the sphere corresponding with the orientations
+        """
+        return self.sphere
+
+    def evaluate_kernel(self, x, y, r, v):
+        return self.k2(x, y, r, v)
```

Comment:
Cython functions cannot be directly called from outside the class (as far as I know) so I needed a Python function as wrapper